nav2_bringup/bringup/params/nav2_params.yaml

```diff
@@ -51,7 +51,7 @@ bt_navigator:
     use_sim_time: True
     global_frame: map
     robot_base_frame: base_link
-    bt_xml_filename: "navigate_w_replanning_and_recovery.xml"
+    default_bt_xml_filename: "set_in_launcher"
```

Comment:
No, this should be valid, there is no assumption that it must be set in launch. The launch configuration exposes the ability to set it on launch and override these parameter file variables, but it is not required (nor should it be).

Reply:
You also need to update this variable in the other 2 parameter files

Reply:
Done

Reply:
But, I am not sure if selecting the default `bt_xml_filename` from launchers are currently hiding here a problem. `navigate_w_replanning_and_recovery.xml` is not a valid filename. If this parameter weren't rewritten in launchers, it wouldn't be a valid filename. 

`bt_xml_filename` is a var taken directly from the parameter, and  `std::ifstream xml_file(bt_xml_filename);` expect a full path. It is not possible for   `std::ifstream` constructor to find the file `navigate_w_replanning_and_recovery.xml` insider behavior_trees directory....

Reply:
Ah good point, ok.